website/server/libs/taskManager.js

```diff
@@ -0,0 +1,170 @@
+import * as Tasks from '../models/task';
+import {
+  BadRequest,
+} from './errors';
+import Bluebird from 'bluebird';
+import _ from 'lodash';
+
+async function _validateTaskAlias (tasks, res) {
+  let tasksWithAliases = tasks.filter(task => task.alias);
+  let aliases = tasksWithAliases.map(task => task.alias);
+
+  // Compares the short names in tasks against
+  // a Set, where values cannot repeat. If the
+  // lengths are different, some name was duplicated
+  if (aliases.length !== [...new Set(aliases)].length) {
+    throw new BadRequest(res.t('taskAliasAlreadyUsed'));
+  }
+
+  await Bluebird.map(tasksWithAliases, (task) => {
+    return task.validateSync();
+  });
+}
+
+
+/**
+ * Creates tasks for a user, challenge or group.
+ *
+ * @param  req  The Express req variable
+ * @param  res  The Express res variable
+ * @param  options  Should be a user, user and challenge to create a challenge task or user and group to create a group task
+ * @return The created tasks
+ */
+export async function createTasks (req, res, options = {}) {
+  let {
+    user,
+    challenge,
+    group,
+  } = options;
+
+  let toSave = Array.isArray(req.body) ? req.body : [req.body];
+
+  toSave = toSave.map(taskData => {
+    // Validate that task.type is valid
+    if (!taskData || Tasks.tasksTypes.indexOf(taskData.type) === -1) throw new BadRequest(res.t('invalidTaskType'));
+
+    let taskType = taskData.type;
+    let newTask = new Tasks[taskType](Tasks.Task.sanitize(taskData));
+
+    if (challenge) {
+      newTask.challenge.id = challenge.id;
+    } else {
```

Comment:
I was thinking we don't depend on not having a task.userId. I think depending on the task.linkTaskId is more reliable and understandable. Challenge has a linkedId as well, so the idea of assuming a task is a challenge task if it doesn't have a user is implicit when the logic could be explicit.